lib/CL/clCreateProgramWithBinary.c

```diff
@@ -123,14 +124,51 @@ POname(clCreateProgramWithBinary)(cl_context                     context,
   program->kernels = NULL;
   program->build_status = CL_BUILD_NONE;
 
+  /* We do no want a binary with different kind of binary in it
```

Comment:
I added a section in design notes about the binary inputs formats allowed by pocl.
See commit : d7f688652df2fa1f21ae955d69d6516f8ff4cdda